# Metagenome analysis using the Kraken software suite

Here we introduce a step-by-step protocol for the Kraken suite, an end-to-end pipeline for the classification, quantification and visualization of metagenomic datasets. Our protocol describes the execution of the Kraken programs, via a sequence of easy-to-use scripts, in two scenarios: (1) quantification of the species in a given metagenomics sample; and (2) detection of a pathogenic agent from a clinical sample taken from a human patient. The protocol, which is executed within 1â€“2 h, is targeted to biologists and clinicians working in microbiome or metagenomics analysis who are familiar with the Unix command-line environment.

## Workflows

We provide two [jupyter notebooks](https://jupyter.org), each desribing one workflow. Jupyter notebooks can be executed for free using Google Colaboratory, which over cloud compute to execute notebooks.

 1. [`Kraken_microbiom.ipynb`](https://github.com/martin-steinegger/kraken-protocol/blob/main/Kraken_microbiom.ipynb) describes the microbiom analysis workflow
 2. [`Kraken_pathogen.ipynb`](https://github.com/martin-steinegger/kraken-protocol/blob/main/Kraken_pathogen.ipynb) describes the pathogen detection workflow

## Publication
[Lu J, Rincon N, Wood D E, Breitwieser F P, Pockrandt C, Langmead B, Salzberg S L, Steinegger M. Metagenome analysis using the Kraken software suite. Nature Protocols, doi: 10.1038/s41596-022-00738-y (2022)](https://www.nature.com/articles/s41596-022-00738-y)
